python/cudf/cudf/core/column/column.py

```diff
@@ -95,7 +242,7 @@ def _concat(cls, objs, dtype=None):
 
         for i, obj in enumerate(objs):
             # Check that all columns are the same type:
-            if not objs[i].is_type_equivalent(head):
+            if not pd.api.types.is_dtype_equal(objs[i].dtype, head.dtype):
```

Comment:
Below we just do `==` instead of using this Pandas API. Is there any reason for the difference here?

Reply:
I think this should be in a follow-up as it looks like it's going to need some digging